frontend/simple/views/Join.vue

```diff
@@ -97,36 +97,41 @@
     </section>
 </template>
 <script>
+import {namespace} from '../js/backend/hapi'
 import * as Events from '../../../shared/events'
 import backend from '../js/backend/'
-import * as db from '../js/database'
-import {latestContractState} from '../js/state'
+import { latestContractState } from '../js/state'
 import L from '../js/translations'
+
 export default {
   name: 'Join',
-  mounted () {
-    this.fetchData()
+  async mounted () {
+    let state = await latestContractState(this.$route.query.groupId)
```

Comment:
Is `this.$route.query.groupId` really the best way to get this parameter? Maybe passing it in via props makes more sense?